lib/chef/resource/windows_user_privilege.rb

```diff
@@ -0,0 +1,149 @@
+                          SeBackupPrivilege
+                          SeChangeNotifyPrivilege
+                          SeSystemtimePrivilege
+                          SeTimeZonePrivilege
+                          SeCreatePagefilePrivilege
+                          SeCreateTokenPrivilege
+                          SeCreateGlobalPrivilege
+                          SeCreatePermanentPrivilege
+                          SeCreateSymbolicLinkPrivilege
+                          SeDebugPrivilege
+                          SeDenyNetworkLogonRight
+                          SeDenyBatchLogonRight
+                          SeDenyServiceLogonRight
+                          SeDenyInteractiveLogonRight
+                          SeDenyRemoteInteractiveLogonRight
+                          SeEnableDelegationPrivilege
+                          SeRemoteShutdownPrivilege
+                          SeAuditPrivilege
+                          SeImpersonatePrivilege
+                          SeIncreaseWorkingSetPrivilege
+                          SeIncreaseBasePriorityPrivilege
+                          SeLoadDriverPrivilege
+                          SeLockMemoryPrivilege
+                          SeBatchLogonRight
+                          SeServiceLogonRight
+                          SeSecurityPrivilege
+                          SeRelabelPrivilege
+                          SeSystemEnvironmentPrivilege
+                          SeManageVolumePrivilege
+                          SeProfileSingleProcessPrivilege
+                          SeSystemProfilePrivilege
+                          SeUndockPrivilege
+                          SeAssignPrimaryTokenPrivilege
+                          SeRestorePrivilege
+                          SeShutdownPrivilege
+                          SeSyncAgentPrivilege
+                          SeTakeOwnershipPrivilege
+                        }
+
+      resource_name :windows_user_privilege
+      description "The windows_user_privilege resource allows to add and set principal (User/Group) to the specified privilege. \n Ref: https://docs.microsoft.com/en-us/windows/security/threat-protection/security-policy-settings/user-rights-assignment"
+
+      introduced "16.0"
+
+      property :principal, String,
+        description: "An optional property to add the user to the given privilege. Use only with add and remove action.",
+        name_property: true
+
+      property :users, Array,
+        description: "An optional property to set the privilege for given users. Use only with set action."
+
+      property :privilege, [Array, String],
+        description: "Privilege to set for users.",
+        required: true,
+        callbacks: {
+           "Option privilege must include any of the: #{privilege_opts}" => lambda {
+             |v| v.is_a?(Array) ? (privilege_opts & v).size == v.size : privilege_opts.include?(v)
+           },
+         }
+
+      property :sensitive, [true, false], default: true
```

Comment:
In chef/chef we use the full class names here:

```suggestion
      property :sensitive, [TrueClass, FalseClass] default: true
```